crypto_kem/kyber768/clean/verify.c

```diff
@@ -1,43 +1,44 @@
 #include <stdint.h>
-#include <string.h>
+#include <stdlib.h>
 
 /*************************************************
- * Name:        verify
- *
- * Description: Compare two arrays for equality in constant time.
- *
- * Arguments:   const unsigned char *a: pointer to first byte array
- *              const unsigned char *b: pointer to second byte array
- *              size_t len:             length of the byte arrays
- *
- * Returns 0 if the byte arrays are equal, 1 otherwise
- **************************************************/
-int PQCLEAN_KYBER768_CLEAN_verify(const unsigned char *a, const unsigned char *b, size_t len) {
+* Name:        verify
+*
+* Description: Compare two arrays for equality in constant time.
+*
+* Arguments:   const unsigned char *a: pointer to first byte array
+*              const unsigned char *b: pointer to second byte array
+*              size_t len:             length of the byte arrays
+*
+* Returns 0 if the byte arrays are equal, 1 otherwise
+**************************************************/
+unsigned char PQCLEAN_KYBER768_CLEAN_verify(const unsigned char *a, const unsigned char *b, size_t len) {
     uint64_t r;
     size_t i;
-    r = 0;
 
+    r = 0;
     for (i = 0; i < len; i++) {
         r |= a[i] ^ b[i];
     }
 
-    r = (-(int64_t)r) >> 63;
-    return (int)r;
+    r = (~r + 1); // Two's complement
```

Comment:
did you want to keep it like this?

Reply:
For the moment, yes. Let's first figure out how to fix the warnings from the MS compiler.